java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.flink.table.types.logical.DecimalType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.LogicalTypeFamily;
+import org.apache.flink.util.StringUtils;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.exporter.PushGateway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.DELETE_ON_SHUTDOWN;
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.EXTRA_LABELS;
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.HOST_URL;
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.JOB_NAME;
+
+/**
+ * A {@link org.apache.flink.streaming.api.functions.sink.SinkFunction} that writes to a Prometheus
+ * PushGateway.
+ *
+ * <p>Note that the SinkFunction only support writing numeric columns.
+ */
+public class PrometheusSinkFunction extends RichSinkFunction<RowData> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PrometheusSinkFunction.class);
+
+    private final ReadableConfig config;
+    private final List<String> columnNames;
+    private final List<String> columnComments;
+    private CollectorRegistry registry;
+    private PushGateway pushGateway;
+    private String jobName;
+    private Map<String, String> extraLabels;
+    private Gauge[] metrics;
+    private final List<DataType> fieldTypes;
+
+    public PrometheusSinkFunction(ReadableConfig config, ResolvedSchema schema) {
+        this.config = config;
+        this.fieldTypes = schema.getColumnDataTypes();
+        this.columnNames =
+                schema.getColumns().stream().map(Column::getName).collect(Collectors.toList());
+        this.columnComments =
+                schema.getColumns().stream()
+                        .map(c -> c.getComment().orElse(c.getName()))
+                        .collect(Collectors.toList());
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+
+        registry = new CollectorRegistry();
+        pushGateway = new PushGateway(config.get(HOST_URL));
+        jobName = config.get(JOB_NAME);
+        extraLabels = parseGroupingKey(config.get(EXTRA_LABELS));
+
+        metrics = new Gauge[columnNames.size()];
+        for (int i = 0; i < columnNames.size(); i++) {
+            final LogicalType logicalType = fieldTypes.get(i).getLogicalType();
+            if (!logicalType.is(LogicalTypeFamily.NUMERIC)) {
+                throw new IOException("PrometheusSink only support numeric data types.");
+            }
+
+            metrics[i] =
+                    Gauge.build()
+                            .name(columnNames.get(i))
```

Comment:
It might be better to decouple column names from prometheus metric names, so that we can write to Prometheus metrics with the same name and different labels.